sympy/matrices/eigen.py

```diff
@@ -752,6 +764,46 @@ def _is_positive_semidefinite_by_minors(M):
     )
 
 
+def _is_positive_semidefinite_by_eigenvalues(M):
+    """Determines if a matrix is positive semidefinite by checking
+    if all the eigenvalues are nonnegative"""
+    return all(eigenvalue.is_nonnegative for eigenvalue in M.eigenvals())
+
+
+def _is_positive_semidefinite_by_cholesky_factorization_with_pivots(M):
```

Comment:
These are too long names that can easily exceed 80 characters.
What about making them shorter like `_is_positive_semidefinite_cholesky`. 
I don't think that it should need much mathematical meaning, especially if these functions are only used internaly.

Reply:
I agree they are a bit verbose. I tend to err on the side of more specificity in my variable names and comments. I shortened them.